include/cantera/zeroD/Wall.h

```diff
@@ -1,35 +1,40 @@
-//! @file Wall.h Header file for class Wall.
+//! @file Wall.h Header file for base class WallBase.
 
 // This file is part of Cantera. See License.txt in the top-level directory or
-// at http://www.cantera.org/license.txt for license and copyright information.
+// at https://cantera.org/license.txt for license and copyright information.
 
 #ifndef CT_WALL_H
 #define CT_WALL_H
 
 #include "cantera/base/ctexceptions.h"
-#include "cantera/numerics/Func1.h"
-#include "cantera/zeroD/ReactorBase.h"
 #include "cantera/zeroD/ReactorSurface.h"
+#include "cantera/zeroD/ReactorBase.h"
+#include "cantera/zeroD/DeviceConst.h"
 
 namespace Cantera
 {
 
 class Kinetics;
 class SurfPhase;
+class Func1;
 
-//! Represents a wall between between two ReactorBase objects.
-/*!
- * Walls can move (changing the volume of the adjacent reactors) and allow heat
- * transfer between reactors.
+/**
+ * Base class for 'walls' (walls, pistons, etc.) connecting reactors.
+ * @ingroup reactor0
  */
-class Wall
+class WallBase
 {
 public:
-    Wall();
+    WallBase();
 
-    virtual ~Wall() {}
-    Wall(const Wall&) = delete;
-    Wall& operator=(const Wall&) = delete;
+    virtual ~WallBase() {}
+    WallBase(const WallBase&) = delete;
+    WallBase& operator=(const WallBase&) = delete;
+
+    //! Return an integer indicating the type of wall
+    int type() {
+        return m_type;
+    }
```

Comment:
I suggest a function with the signature `virtual std::string type() const`, like in class `ThermoPhase`.

Reply:
as mentioned above, I used `Reactor` (and also `FlowDevice`) classes as a template and didn't want to go a different direction. It's easy to change if so desired.

Reply:
At close inspection, this involves major changes in CLIB - which builds on the constructor  `int wall_new(int type)`, - potentially breaking backwards compatibility. I propose to defer elimination/deprecation of magic numbers in a separate effort.

Reply:
Except that `wall_new` currently doesn't use the argument passed in since there is only one type of wall. So, no one can be using that argument to actually do anything.

I agree that figuring out how to transition away from the existing magic numbers is best left to another PR, but we can make the task at least a little simpler if we avoid putting things in our own way. So, while I see the temporary need for being able to assign an integer constant to a custom `Wall` class via the `reg_type` function, I'd rather we use the name `type` here for the string constant, and something else for the integer constant (if that's even really necessary).

Reply:
The way I implemented things, instantiation of objects via `WallFactory` is an exact replica of objects created via `ReactorFactory` calls (note that in `ctreactor.cpp` I changed calls `wall_new` and `flowdev_new` to work exactly like `reactor_new`, i.e. the argument is actually used). 

I'd much prefer to use the _old_ coding style for the time being and change everything in a separate PR. In other words, since there are precedents for `int FlowDevice::type()` and `int ReactorBase::type()` signatures, I am hesitant to introduce a `std::string WallBase::type()`.

Reply:
OK, the clarification makes sense. Since this is currently not used by any call, changing signature is trivial.